main.tf

```diff
@@ -515,6 +535,13 @@ resource "ibm_container_addons" "addons" {
       version = addons.value
     }
   }
+  # OCP AI addon Validations
+  lifecycle {
```

Comment:
Is there a reason why this is checked? we do not do this for any of the other add-ons, why would we do it for this particular one?

Also, this lifecycle is for the ENTIRE addons resource block. So this condition will block all of the add-ons, not just ai.

Reply:
Thank you for reviewing this.  Actually I should remove it. 

The thought behind this was if existing cluster already has ocp-ai add on installed and if user tries to install ocp-ai addon on a cluster that already has ocp-ai addon, this should not allow the user. 
But the existing cluster use case is not valid at the moment as we are creating a new cluster.

Reply:
This logic really applies to all of the add-ons possible though, not just this ai one, so no need to single it out. If we were to do a condition like this, we should do it for all addons.

However there is an issue to resolve in this approach, in that the lifecycle condition won't just stop that one add-on from being installed, I believe it would stop ALL add-ons from being installed if just one of them causes the condition to fail.

We do give the end user the ability to manage all addons with terraform, or have terraform only manage the ones in the input list, which might be able to be used for similar effect.

Reply:
Sure, noted. This will be very helpful when working on existing clusters part too.